### ISSUE TITLE
r.now and changefeeds

### ISSUE THREAD

**Issue by AtnNn** [rethinkdb/rethinkdb#3583]
The current behaviour of `r.now` makes it less useful in the context of changefeeds.

If it was not constant, it would allow adding timestamps to the results and filtering based on the current time:

```
.changes().map(lambda row: [r.now(), row])
.changes().filter(lambda row: row.end_time > r.now())
```

**Comment by Timer** [comment]
This would seem like a natural progression resulting from the addition of change feeds -- I would like to see this implemented as I need it for my application.

**Comment by mlucy** [comment]
Changing the behavior of `r.now` in that way would make it non-deterministic, which would break some people's queries since it could no longer be used in atomic updates.  Maybe we could add an optarg to change the behavior?

**Comment by shadowcodex** [comment]
Was wondering if this has been addressed yet. Here is my use case:

Example: Watching user action records on a conditional of (if action was less than 5 minutes ago). Pull a query that matches that, and if as time passes no records match anymore to emit an update. Thus easily showing what users are “Currently Active" without having to do a polling situation.

**Comment by thelinuxlich** [comment]
+1 and I think for this to be really useful it would need to send changefeed events when the document leaves the time window({new_val: null, old_val: { ... }})

**Comment by tatsujin1** [comment]
Yes, filtering using the current time would be very useful in some situations I have!

But I guess there would need to be two `r.now()`; one `r.query_start_now()` (i.e. the current one), and one `r.current_now()`, if you know what I mean?   :)

One simple case I have, or want to have, because doing it without this feature is complicated :), is this:
```
for change in table.filter(lambda doc: r.current_now() - doc['modtime'] > 600).changes().run():
    stuff...
```
That is, get the document once it has been "stable" (unmodified) for 10 minutes.
Completely awesome!  Would this be possible?    Pleeeeease? :D

**Comment by danielmewes** [comment]
@tatsujin1 Good point. We indeed need to figure out a nice way for distinguishing the two variants since both can be useful as you say. Maybe with a separate term, or with the optarg (I think the former might be easier to implement).

**Comment by thelinuxlich** [comment]
A third option(probably crazy), would be a lazy wrapper:

```
.changes().filter(lambda row: row.end_time > r.lazy(r.now()))
```

**Comment by tatsujin1** [comment]
I like this.
In theory such a `lazy` operator might be useful in other situations as well. I have no idea, currently, what these might be, however. :)

Also, this would supposedly hide whatever "ugly" solution is selected (`current_now()` or `r.now(dynamic=True)`, etc).

**Comment by tatsujin1** [comment]
I realized  (a while back) that this feature would make my suggestions on #746 and #4008 about non-deleting variants a lot easier and also more explicit (i.e. better) than using TTL-like functionality. At least in most of my cases.

For example:
`table.filter(r.lazy(r.now()) >= r.row['expires_at']).changes()`
vs (using a TTLified index):
`table.changes()`

What I guess I'm trying to say is that this feature is more generic :)

**Comment by blindman2k** [comment]
+1 for me. I'm looking for a feed of recently updated documents. I can catch the new documents easily but timing out old documents is tricky at scale.

**Comment by thelinuxlich** [comment]
Maybe this can be solved by r.write_time(#6396), which is merged for 2.4?
